lpd/tests/test_models.py

```diff
@@ -1,11 +1,317 @@
+import random
+
 from django.test import TestCase
 
-from lpd.models import LearnerProfileDashboard
+from lpd.constants import QuestionTypes
+from lpd.models import (
+    AnswerOption,
+    KnowledgeComponent,
+    LearnerProfileDashboard,
+    QuantitativeAnswer,
+    RankingQuestion,
+    Score
+)
+from lpd.tests.factories import (
+    LearnerProfileDashboardFactory,
+    LikertScaleQuestionFactory,
+    MultipleChoiceQuestionFactory,
+    QualitativeAnswerFactory,
+    QualitativeQuestionFactory,
+    RankingQuestionFactory,
+    SectionFactory,
+    UserFactory
+)
+
+
+# Globals
+
+QUALITATIVE_QUESTION_FACTORIES = [
+    QualitativeQuestionFactory,
+]
+QUANTITATIVE_QUESTION_FACTORIES = [
+    MultipleChoiceQuestionFactory,
+    RankingQuestionFactory,
+    LikertScaleQuestionFactory
+]
+QUESTION_FACTORIES = QUALITATIVE_QUESTION_FACTORIES + QUANTITATIVE_QUESTION_FACTORIES
+
 
+# Classes
 
 class LearnerProfileDashboardTests(TestCase):
     """LearnerProfileDashboard model tests."""
 
     def test_str(self):
+        """
+        Test string representation of `LearnerProfileDashboard` model.
+        """
         lpd = LearnerProfileDashboard(name='Empty LPD')
         self.assertEquals(str(lpd), 'Empty LPD')
+
+
+class SectionTests(TestCase):
+    """Section model tests."""
+
+    def setUp(self):
+        self.lpd = LearnerProfileDashboardFactory(name='Test LPD')
+
+    def test_str(self):
+        """
+        Test string representation of `Section` model.
+        """
+        section = SectionFactory(lpd=self.lpd, title='Basic information')
+        self.assertEquals(str(section), 'Section 1: Basic information')
+
+    def test_questions(self):
+        """
+        Test that `questions` property returns questions belonging to `Section` in appropriate order.
+        """
+        section = SectionFactory(lpd=self.lpd, title='Details, Details, Details')
+        questions = []
+        question_numbers = sorted(random.sample(range(1, 100), 20))
+        for unused in range(5):
+            for question_factory in QUESTION_FACTORIES:
+                question = question_factory(section=section, number=question_numbers.pop(0))
```

Comment:
Just to confirm: this will raise exception if we add another factory to `QUESTION_FACTORIES`
but that's intended, right?

Otherwise it might be better to exchange `20` in `random.sample()` for:

```python
5 * len(QUESTION_FACTORIES)
```

Reply:
@tomaszgy It's not intended, I just don't expect that we will add more question types any time soon (so we won't need any additional `QUESTION_FACTORIES`) 🙂 

But it won't hurt to make this more robust in any case, so I've done that in 6187ffdf84edce67f833865819948493b8dd945c.